### PR TITLE
Add configuration back

### PR DESCRIPTION
**Pull request by sagikazarmark**
@dbu Better now?

Two things I am not completely sure about:
- Currently the service is defined in the extension. I didn't want to inject the config as parameters. Does it violate any Symfony Best Practice?
- If translations are turned of, you can and you HAVE TO override all the labels, otherwise you get trans keys. Any ideas to make it better?

### CODE REVIEW

File: DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        $rootNode = $treeBuilder->root('indigo_cookie_consent');
+
+        $rootNode
+            ->children()
+                ->arrayNode('options')
+                    ->addDefaultsIfNotSet()
+                    ->children()
+                        ->scalarNode('message')->defaultValue('message')->end()
+                        ->scalarNode('dismiss')->defaultValue('dismiss')->end()
+                        ->scalarNode('learnMore')->defaultValue('learn_more')->end()
```
Comment (dbu) [inline DependencyInjection/Configuration.php]: symfony configuration values usually use underscores rather than camelCase

Comment (sagikazarmark) [inline DependencyInjection/Configuration.php]: I know, but this options config is passed to the plugin as is and it wants it camelCased. I didn't want to add normalization logic. It is mentioned in the docs that the options config accepts the same parameters as the plugin, link placed.

Comment (dbu) [inline DependencyInjection/Configuration.php]: but with the translation namespace existing, i am not sure if its even worth the effort to allow configuring the translation keys - its unique enough and the user can not use some other translation they already have as it must be in the namespace anyways.

File: DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+                        ->scalarNode('link')->end()
+                        ->scalarNode('container')->end()
+                        ->scalarNode('theme')->end()
```
Comment (dbu) [inline DependencyInjection/Configuration.php:26]: i would add some `info()` to these configurations - its not self-evident what container or theme are. maybe also just drop them, if they are for the template. the user can overwrite the twig template in app/Resources/IndigophpCookieConsentBundle/twig/ if he needs to change the layout.

File: DependencyInjection/Configuration.php (hunk 1 above)
Comment (sagikazarmark) [inline DependencyInjection/Configuration.php]: If you disable translation, you can pass any text here which is then displayed. However this is one of the things I am not sure about: we have translation, should we allow to override labels in the config? Or force users to override translation domains instead?

File: DependencyInjection/Configuration.php (hunk 2 above)
Comment (sagikazarmark) [inline DependencyInjection/Configuration.php:26]: These options are passed to the plugin. Maybe adding info makes sense, but it is mentioned in the docs that these are configuration keys of the plugin.

File: DependencyInjection/Configuration.php (hunk 1 above)
Comment (dbu) [inline DependencyInjection/Configuration.php]: ah, i should have read the PR description, sorry.

hm, i would say screw them :-) i would put the translation calls into the twig template, not allow any configuration for that and tell users to override the template if they disable translations. keeps things a lot simpler and imho even for mono-language websites you should use translations to separate semantical meaning from actual translation. it helps knowing what text snippets are used on a website.

**Comment by dbu** [comment]
> Currently the service is defined in the extension. I didn't want to inject the config as parameters. Does it violate any Symfony Best Practice?

i am not exactly sure if there are rules about that. but i always see setting the configuration in a container parameter and have the service definition. imho it would be the expected way, rather than define the whole service in the extension class. there is no harm in having that container parameter imho. and a parameter can also contain an array, so you would not need to change the twig extension signature.

File: DependencyInjection/Configuration.php (hunk 1 above)
Comment (sagikazarmark) [inline DependencyInjection/Configuration.php]: Thanks.

Currently the translation is done in the extension. The template just receives an array of options, converts it to json and passes them to the plugin.

File: DependencyInjection/Configuration.php (hunk 2 above)
Comment (dbu) [inline DependencyInjection/Configuration.php:26]: ah i see, duplication of documentation is an issue of course. i'd try to reduce the amount of things to configure in the first place, i think.

File: DependencyInjection/Configuration.php (hunk 1 above)
Comment (dbu) [inline DependencyInjection/Configuration.php]: ah right, i see. then i would just hardcode the translations in the twig extension. if everything is different, people can also just use the javascript directly without the bundle - there is a point where that becomes simpler than configuring everything.

File: DependencyInjection/Configuration.php (hunk 2 above)
Comment (dbu) [inline DependencyInjection/Configuration.php:26]: oh, and they go to the javascript lib as well? then i guess its ok like this - maybe add an info to each saying that this parameter is passed to the js lib and documented there?

Comment (sagikazarmark) [inline DependencyInjection/Configuration.php:26]: Agree. Translations are actually the only feature this Bundle adds.

Comment (sagikazarmark) [inline DependencyInjection/Configuration.php:26]: Can I add documentation to the array node? Or it won't be displayed?

Comment (dbu) [inline DependencyInjection/Configuration.php:26]: i think you can, yes. there is a console command to display the information of a bundle config if you want to make sure.

**Pull request merged**